### DISCUSSION TITLE
Tuple to Object

### DISCUSSION THREAD

**Discussion by michaeljsalo** [type-challenges/type-challenges#8443] [Q&A]
The solutions for 'Tuple to Object' problem apply a type map that looks like this:

`[key in T[number]]: key`

I understand how the mapping works, but what I didn't know about is T[number].  Is there TypeScript documentation that explains the T[number] expression for arrays.

Also, would object types have an equivalent expression?  If T was an object type with string keys and string values, does this mean I could use T[string] to get a union of all the strings?

**Comment by JonghwanWon** [comment]
I hope [document](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html) will help you.

**Comment by emayom** [comment]
If T is an object type **containing only string keys** and string values, you can make use of T[string], which is a form of indexed access type!

**Comment by Tammy648** [comment]
Hi, I’m currently learning TypeScript and came across this discussion while starting to learn about tuples and arrays. I understand the basic idea of mapping types, but the use of T[number] is still a bit confusing to me, especially how it creates a union of values from an array. I think it would really help if there was a simpler example or explanation showing how this works step by step.
Thanks!
